workitem/workitem_repository.go

```diff
@@ -687,39 +690,58 @@ func (r *GormWorkItemRepository) Fetch(ctx context.Context, spaceID uuid.UUID, c
 // 		WHERE (iterations.space_id = '33406de1-25f1-4969-bcec-88f29d0a7de3'
 // 		and work_items.deleted_at IS NULL) GROUP BY IterationId
 func (r *GormWorkItemRepository) GetCountsPerIteration(ctx context.Context, spaceID uuid.UUID) (map[string]WICountsPerIteration, error) {
-	var res []WICountsPerIteration
-	db := r.db.Table("work_items").Select(`iterations.id as IterationId, count(*) as Total,
-				count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed`).Joins(`left join iterations
-				on fields@> concat('{"system.iteration": "', iterations.id, '"}')::jsonb`).Where(`iterations.space_id = ?
-				and work_items.deleted_at IS NULL`, spaceID).Group(`IterationId`).Scan(&res)
+	var itrs []*iteration.Iteration
+	db := r.db.Debug()
+	db = db.Table("iterations").Select("*").Where("space_id = ? AND deleted_at IS NULL", spaceID).Scan(&itrs)
 	if db.Error != nil {
 		return nil, errors.NewInternalError(db.Error.Error())
 	}
 	countsMap := map[string]WICountsPerIteration{}
-	for _, iterationWithCount := range res {
-		countsMap[iterationWithCount.IterationId] = iterationWithCount
+	for _, itr := range itrs {
```

Comment:
Thanks @aslakknutsen ,
Yes, now I have tried to make it better than before:
Strategy - 

get counts for every iteration in that space in a map

Closed | Total | IterationID
-----| ------- | -------------
3 | 5 | i1
5 | 15 | i2

Add 0 count rows in above result in Go code. (I know I can do it in the query itself ,but I was unable to add 0 count rows in above query itself.)

then, fetch children list for every iteration in given space

Children | IterationID
----- | -------------
{i1, i2, i3} | i0
{i5, i6} | i1

Now iterate over first result 
- for every iteration:
   - check if it has children
             - if children available then iterate over children and sum up the counts 

So now, in total 3 queries (irrespective of no. of iterations) are fired in this method, and rest of the logic is done in GO.
I tried a lot to put all logic in one query but failed several times hence the delay in PR (but I was busy with other tasks :) )